canonical/lxd-ui
# Improve networking

## Done

- address various issues around network creation/editing discovered during in person review. Notes are in [this doc](https://docs.google.com/document/d/1BehzsvBhTVXrSrlSjFUoo7eMppzwmNLNfYQ_aDhfgi4/edit#heading=h.o2hu4ulgsezd)
- remove fan network type (simplifies bridge network handling)
- streamline copy on network forms
- fix ovn uplink network list to respect restricted.networks.uplinks setting for project, when that is unset, the non-ovn networks from the default project should be displayed in the list.

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - created a bridge network
    - edit a bridge network
    - if ovn available, create / edit an ovn network
    - if ovn available, setup a project with restricted.networks.uplinks set. Enter some network names from the default project into  project configuration > Allow custom restrictions on a project level > Restrictions > Networks > Network uplinks. Only networks explicitly listed should be available as uplink, when creating an ovn network in that project.
